generators/beamAndVertexGenerator.cc

```diff
@@ -66,11 +103,16 @@ bool beamAndVertexGenerator::loadBeamFile(const string& beamFileName)
 		_beamTree = NULL;
 		return false;
 	}
-	_beamTree->SetBranchAddress("vertex_x_position", &_vertexX.first);
-	_beamTree->SetBranchAddress("vertex_y_position", &_vertexY.first);
-	_beamTree->SetBranchAddress("beam_momentum_x", &_beamMomentumX.first);
-	_beamTree->SetBranchAddress("beam_momentum_y", &_beamMomentumY.first);
-	_beamTree->SetBranchAddress("beam_momentum_z", &_beamMomentumZ.first);
+	_beamTree->SetBranchAddress("vertex_x_position", &_vertexX);
+	_beamTree->SetBranchAddress("vertex_y_position", &_vertexY);
+	if(vertexZPosition){
+		_beamTree->SetBranchAddress("vertex_z_position", &_vertexZ);
+	} else {
+		_vertexZ = 0.0; // per default the beam parameters are given at Z=0
+	}
+	_beamTree->SetBranchAddress("beam_momentum_x", &_beamMomentumX);
+	_beamTree->SetBranchAddress("beam_momentum_y", &_beamMomentumY);
+	_beamTree->SetBranchAddress("beam_momentum_z", &_beamMomentumZ);
 	TBranch* vertexXPositionSigma = _beamTree->FindBranch("vertex_x_position_sigma");
```

Comment:
question of taste: maybe the code would be easier to follow if it would be organized in 3 blocks:
i) assignment of all branch pointers
ii) check of required and optional branches
iii) connection of leafs